### PR TITLE
[DCS]: v2 instance

### PR DESCRIPTION
**Pull request by artem-lifshits**
## Summary of the Pull Request
New DCSv2 instance resource.

## PR Checklist

* [x] Refers to: #2253
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccDcsInstancesV2_basic
=== PAUSE TestAccDcsInstancesV2_basic
=== CONT  TestAccDcsInstancesV2_basic
--- PASS: TestAccDcsInstancesV2_basic (138.68s)
=== RUN   TestAccDcsInstancesV2_privateIPs
=== PAUSE TestAccDcsInstancesV2_privateIPs
=== CONT  TestAccDcsInstancesV2_privateIPs
--- PASS: TestAccDcsInstancesV2_privateIPs (113.86s)
=== RUN   TestAccDcsInstancesV2_basicSingleInstance
=== PAUSE TestAccDcsInstancesV2_basicSingleInstance
=== CONT  TestAccDcsInstancesV2_basicSingleInstance
--- PASS: TestAccDcsInstancesV2_basicSingleInstance (101.24s)
=== RUN   TestAccDcsInstancesV2_basicEngineV3Instance
=== PAUSE TestAccDcsInstancesV2_basicEngineV3Instance
=== CONT  TestAccDcsInstancesV2_basicEngineV3Instance
--- PASS: TestAccDcsInstancesV2_basicEngineV3Instance (410.58s)
=== RUN   TestAccDcsInstancesV2_Whitelist
=== PAUSE TestAccDcsInstancesV2_Whitelist
=== CONT  TestAccDcsInstancesV2_Whitelist
--- PASS: TestAccDcsInstancesV2_Whitelist (223.31s)
=== RUN   TestAccDcsInstancesV2_SSL
=== PAUSE TestAccDcsInstancesV2_SSL
=== CONT  TestAccDcsInstancesV2_SSL
--- PASS: TestAccDcsInstancesV2_SSL (189.35s)
=== RUN   TestAccDCSInstanceV2_importBasic
--- PASS: TestAccDCSInstanceV2_importBasic (120.75s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

**Review by sgmv — DISMISSED** [review]

File: opentelekomcloud/acceptance/dcs/resource_opentelekomcloud_dcs_instance_v2_test.go [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { common.TestAccPreCheck(t) },
+		ProviderFactories: common.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckDcsV2InstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDcsV2InstanceBasic(instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDcsV2InstanceExists(dcsV2InstanceName, dcsInstance),
```
Comment (anton-sidelnikov) [inline opentelekomcloud/acceptance/dcs/resource_opentelekomcloud_dcs_instance_v2_test.go:47]: could you use here resource_check from common?

File: opentelekomcloud/services/dcs/resource_opentelekomcloud_dcs_instance_v2.go [hunk 2]
```diff
@@ -0,0 +1,1231 @@
...
+			"template_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/dcs/resource_opentelekomcloud_dcs_instance_v2.go]: remove please

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Review by Aloento — DISMISSED** [review]

File: opentelekomcloud/acceptance/dcs/resource_opentelekomcloud_dcs_instance_v2_test.go (hunk 1 above)
Comment (artem-lifshits) [reply opentelekomcloud/acceptance/dcs/resource_opentelekomcloud_dcs_instance_v2_test.go:47]: done

File: opentelekomcloud/services/dcs/resource_opentelekomcloud_dcs_instance_v2.go (hunk 2 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/dcs/resource_opentelekomcloud_dcs_instance_v2.go]: done

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**